File: php/features.php
<?php
declare(strict_types=1);

// Covid19Data SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class Covid19DataFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new Covid19DataBaseFeature();
            case "test":
                return new Covid19DataTestFeature();
            default:
                return new Covid19DataBaseFeature();
        }
    }
}

File: py/features.py
# Covid19Data SDK feature factory

from feature.base_feature import Covid19DataBaseFeature
from feature.test_feature import Covid19DataTestFeature


def _make_feature(name):
    features = {
        "base": lambda: Covid19DataBaseFeature(),
        "test": lambda: Covid19DataTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
